api/graphql/reservation_units/reservation_unit_filtersets.py

```diff
@@ -196,10 +195,14 @@ def get_min_persons_lte(self, qs, property, value):
         return qs.filter(filters)
 
     def get_is_visible(self, qs, property, value):
-        today = datetime.datetime.now(tz=get_default_timezone())
+        now = timezone.now()
         qs = qs.filter(is_draft=False, is_archived=False)
-        published = (Q(publish_begins__lte=today) | Q(publish_begins__isnull=True)) & (
-            Q(publish_ends__gt=today) | Q(publish_ends__isnull=True)
+        published = (Q(publish_begins__lte=now) | Q(publish_begins__isnull=True)) & (
+            Q(publish_ends__gt=now)
+            | Q(publish_ends__isnull=True)
+            | Q(
+                publish_ends__lt=F("publish_begins")
+            )  # When end is before begin and begin is today or before it is considered as (re)published.
```

Comment:
I think this comment is now slightly inaccurate because if begin is later today the unit is not considered as republished right now.